### PR TITLE
🐛 [amp-next-page] fix `ampdoc.getUrl()` is empty on next page

### LINKED ISSUE #33886: `ampdoc.getUrl()` is empty on page loaded by `<amp-next-page>`
When new page append, Addthis buttons not sharing current url. It is sharing `https://s7.addthis.com/dc/amp-addthis.html?_amp_=2104022034000`. And share counts not showing true results.

### PR DESCRIPTION
**Pull request by dmanek**
Fixes #33886

After (using log points in CDT): https://user-images.githubusercontent.com/506183/116159441-1827ac80-a6a5-11eb-8ade-c44a356ae9f8.mp4

### CODE REVIEW

File: extensions/amp-next-page/1.0/service.js [hunk 1]
```diff
@@ -558,9 +558,10 @@ export class NextPageService {
    * @param {!Page} page
    * @param {!Document} content
    * @param {boolean=} force
+   * @param {string=} url
```
Comment (alanorozco) [inline extensions/amp-next-page/1.0/service.js:561]:
```suggestion
```

File: extensions/amp-next-page/1.0/service.js [hunk 2]
```diff
@@ -558,9 +558,10 @@ export class NextPageService {
...
    * @return {!Promise<?../../../src/runtime.ShadowDoc>}
    */
-  attachDocumentToPage(page, content, force = false) {
+  attachDocumentToPage(page, content, force = false, url = '') {
```
Comment (alanorozco) [inline extensions/amp-next-page/1.0/service.js:564]:
```suggestion
  attachDocumentToPage(page, content, force = false) {
```

File: extensions/amp-next-page/1.0/service.js [hunk 3]
```diff
@@ -600,7 +601,7 @@ export class NextPageService {
       const amp = this.multidocManager_.attachShadowDoc(
         shadowRoot,
         content,
-        '',
+        url,
```
Comment (alanorozco) [inline extensions/amp-next-page/1.0/service.js:604]: ...and this PR becomes a one line change!

```suggestion
        page.url,
```

File: extensions/amp-next-page/1.0/page.js [hunk 4]
```diff
@@ -253,7 +253,8 @@ export class Page {
       .attachDocumentToPage(
         this /** page */,
         /** @type {!Document} */ (devAssert(this.content_)),
-        this.is(PageState.PAUSED) /** force */
+        this.is(PageState.PAUSED) /** force */,
+        this.url_
```
Comment (alanorozco) [inline extensions/amp-next-page/1.0/page.js:257]: It looks like the `Page` object has a `url` getter, so we don't need this arg.
```suggestion
```

File: extensions/amp-next-page/1.0/service.js (hunk 3 above)
Comment (dmanek) [reply extensions/amp-next-page/1.0/service.js]: nice!

**Review by alanorozco — APPROVED** [review]

**Pull request merged**